File: core/java/src/main/java/hara/lang/protocol/IPeekLast.java
package hara.lang.protocol;

import hara.lang.declaration.HaraMethod;
import hara.lang.declaration.HaraProtocolBinding;

@HaraProtocolBinding(namespace = "std.protocol.ipeeklast", name = "IPeekLast")
public interface IPeekLast<E> {
  @HaraMethod(value = "peek-last", arity = 1)
  E peekLast();
}

File: core/rust/src/lang/protocol/ipeeklast.rs
use hara_protocol_macros::hara_protocol;

#[hara_protocol(namespace = "std.protocol.ipeeklast", name = "IPeekLast")]
pub trait IPeekLast<E> {
    #[hara_method(value = "peek-last", arity = 1)]
    fn peek_last(&self) -> Option<E>;
}
